Duplicati/Library/Backend/TencentCOS/COSBackend.cs

```diff
@@ -0,0 +1,431 @@
+                string bucket = _cosOptions.Bucket;
+
+                GetBucketRequest request = new GetBucketRequest(bucket);
+
+                //设置签名有效时长
+                request.SetSign(TimeUtils.GetCurrentTime(TimeUnit.SECONDS), 600);
+
+                if (!string.IsNullOrEmpty(filename))
+                    request.SetMarker(filename);
+
+                //获取 m_prefix/ 下的对象
+                if (!string.IsNullOrEmpty(m_prefix))
+                    request.SetPrefix(m_prefix);
+
+                //执行请求
+                GetBucketResult result = _cosXml.GetBucket(request);
+
+                //bucket的相关信息
+                ListBucket info = result.listBucket;
+
+                isTruncated = result.listBucket.isTruncated;
+                filename = result.listBucket.nextMarker;
+
+                foreach (var item in info.contentsList)
+                {
+                    var last = DateTime.Parse(item.lastModified);
+
+                    var fileName = item.key;
+                    if (!string.IsNullOrWhiteSpace(m_prefix))
+                    {
+                        fileName = fileName.Substring(m_prefix.Length);
+
+                        if (fileName.StartsWith("/", StringComparison.Ordinal))
+                        {
+                            fileName = fileName.Trim('/').Trim('/');
+                        }
+                    }
+
+                    yield return new FileEntry(fileName, item.size, last, last);
+                }
+            }
+        }
+
+        public Task PutAsync(string remotename, string filename, CancellationToken cancelToken)
+        {
+            using (FileStream fs = File.Open(filename, FileMode.Open, FileAccess.Read, FileShare.Read))
+                return PutAsync(remotename, fs, cancelToken);
+        }
+
+        public void Get(string remotename, string filename)
+        {
+            using (var fs = File.Open(filename, FileMode.Create, FileAccess.Write, FileShare.None))
+                Get(remotename, fs);
+        }
+
+        public void Delete(string remotename)
+        {
+            try
+            {
+                _cosXml = GetCosXml();
+                string bucket = _cosOptions.Bucket;
+                string key = GetFullKey(remotename);
+
+                //string bucket = "examplebucket-1250000000"; //存储桶，格式：BucketName-APPID
+                //string key = "exampleobject"; //对象在存储桶中的位置，即称对象键
+                DeleteObjectRequest request = new DeleteObjectRequest(bucket, key);
+                //设置签名有效时长
+                request.SetSign(TimeUtils.GetCurrentTime(TimeUnit.SECONDS), 600);
+                //执行请求
+                DeleteObjectResult result = _cosXml.DeleteObject(request);
+                //请求成功
+                Console.WriteLine(result.GetResultInfo());
+            }
+            catch (COSXML.CosException.CosClientException clientEx)
+            {
+                //请求失败
+                Console.WriteLine("CosClientException: " + clientEx);
```

Comment:
We should avoid using `Console.WriteLine`.  Instead, use `Logging.Log.WriteErrorMessage` or just leave the exception unhandled.